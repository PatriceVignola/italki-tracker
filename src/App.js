/**
 * @prettier
 * @flow
 */

import React from 'react';
import {Provider} from 'react-redux';
import {BrowserRouter, Route, Switch} from 'react-router-dom';
import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';
import type {Store} from 'redux';

import reducers from './reducers';
import LeftDrawer from './components/LeftDrawer';
import TopAppBar from './components/TopAppBar';
import StudentListScreen from './components/ConnectedStudentList';
import type {Action} from './actions/types';

const initialState = {};

type State = {
  drawerOpen: boolean,
};

class App extends React.Component<{}, State> {
  store: Store<Object, Action>;
  handleDrawerOpen: () => void;
  handleDrawerClose: () => void;

  constructor() {
    super();

    this.store = createStore(reducers, initialState, applyMiddleware(thunk));

    this.state = {
      drawerOpen: false,
    };

    this.handleDrawerOpen = this.handleDrawerOpen.bind(this);
    this.handleDrawerClose = this.handleDrawerClose.bind(this);
  }

  handleDrawerOpen() {
    this.setState({drawerOpen: true});
  }

  handleDrawerClose() {
    this.setState({drawerOpen: false});
  }

  render() {
    const baseUrl: string = (process.env.PUBLIC_URL: any);

    return (
      <Provider store={this.store}>
        <BrowserRouter basename={baseUrl}>
          <div>
            <TopAppBar onHamburgerClick={this.handleDrawerOpen} />
            <LeftDrawer
              open={this.state.drawerOpen}
              onClose={this.handleDrawerClose}
            />
            <Switch>
              <Route exact path="/" component={StudentListScreen} />
            </Switch>
          </div>
        </BrowserRouter>
      </Provider>
    );
  }
}

export default App;
