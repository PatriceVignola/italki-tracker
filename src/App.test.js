/**
 * @prettier
 * @flow
 */

import React from 'react';
import {shallow} from 'enzyme';

import App from './App';

describe('App', () => {
  it('renders correctly when PUBLIC_URL is set', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
