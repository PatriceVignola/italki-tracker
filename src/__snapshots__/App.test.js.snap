// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly when PUBLIC_URL is set 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BrowserRouter>
    <div>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
      </Switch>
    </div>
  </BrowserRouter>
</Provider>
`;
